### PR TITLE
best order for proposals to utils

### PR DESCRIPTION
**Pull request by wwongkamjan**
I have updated utils / dipnet/ realpolitik

### CODE REVIEW

File: src/baseline_bots/utils.py [hunk 1]
```diff
@@ -298,6 +298,38 @@ def get_state_value(bot, game, power_name):
         game.process()
     return len(game.get_centers(power_name))
 
+@gen.coroutine
+def get_best_orders(bot, proposal_order, shared_order):
+    # input: sender power, dipnet_order + incoming proposals {power: [orders]}, shared_orders, Diplomacy game
```
Comment (trigaten) [inline src/baseline_bots/utils.py:303]: Can you put a docstring here ("""comments""").

File: src/baseline_bots/utils.py [hunk 2]
```diff
@@ -298,6 +298,38 @@ def get_state_value(bot, game, power_name):
         game.process()
     return len(game.get_centers(power_name))
 
+@gen.coroutine
+def get_best_orders(bot, proposal_order, shared_order):
```
Comment (trigaten) [inline src/baseline_bots/utils.py:302]: Can you put type hints for the parameters, e.g. `bot:BaselineBot`

File: src/baseline_bots/utils.py [hunk 3]
```diff
@@ -298,6 +298,38 @@ def get_state_value(bot, game, power_name):
...
+    }
+    for proposer, unit_orders in proposal_order.items():
+        if unit_orders:
+            proposed = True
+            simulated_game = bot.game.__deepcopy__(None)
+            unit_orders = get_non_aggressive_orders(
+                unit_orders, bot.power_name, bot.game
+            )
+            # print('from: ', proposer)
+            # print(orders)
```
Comment (trigaten) [inline src/baseline_bots/utils.py:318]: Please remove commented out code

**Review by trigaten — CHANGES_REQUESTED** [review]
Looking good! Just a few changes needed. Don't forget to run the makefile for code styling.

**Comment by wwongkamjan** [comment]
I have a problem import BaselineBot in utils.py

this is for hints in below function
`def get_best_orders(bot: BaselineBot, proposal_order: dict, shared_order: dict):`

**Comment by trigaten** [comment]
@wwongkamjan can you provide the error you are receiving? You may need to use Dict instead of dict. Also, you can run tests locally via pytest tests/ in terminal.

File: src/baseline_bots/utils.py [hunk 4]
```diff
@@ -299,6 +300,36 @@ def get_state_value(bot, game, power_name):
     return len(game.get_centers(power_name))
 
 
+@gen.coroutine
+def get_best_orders(bot: BaselineBot, proposal_order: dict, shared_order: dict):
+    """
+    input: sender power, dipnet_order + incoming proposals {power: [orders]}, shared_orders, Diplomacy game
+    output: [orders] a list of orders (with best value)
+                # for each xdo order set (max at 6 for now) -> simulate worlds by execute all of shared orders + xdo order set
```
Comment (trigaten) [inline src/baseline_bots/utils.py:308]: Why is there a # comment inside the docstring?

Comment (wwongkamjan) [reply src/baseline_bots/utils.py:308]: oops I forgot to remove

**Comment by wwongkamjan** [comment]
This is an error I got. 
 
 ImportError: cannot import name 'BaselineBot' from 'baseline_bots.bots.baseline_bot' (/home/runner/work/baseline_bots/baseline_bots/src/baseline_bots/bots/baseline_bot.py)

**Comment by wwongkamjan** [comment]
https://github.com/ALLAN-DIP/baseline_bots/actions/runs/3145380014/jobs/5112589351 I think you can also see it here?

**Comment by trigaten** [comment]
> This is an error I got.
> 
> ImportError: cannot import name 'BaselineBot' from 'baseline_bots.bots.baseline_bot' (/home/runner/work/baseline_bots/baseline_bots/src/baseline_bots/bots/baseline_bot.py)

Try `from baseline_bots import BaselineBot`. Look at how other files import `BaselineBot`

**Comment by wwongkamjan** [comment]
from baseline_bots import BaselineBot
ImportError: cannot import name 'BaselineBot' from 'baseline_bots' (/home/runner/work/baseline_bots/baseline_bots/src/baseline_bots/__init__.py)

https://github.com/ALLAN-DIP/baseline_bots/actions/runs/3145498107/jobs/5112862819

It didn't work :(

**Comment by trigaten** [comment]
@wwongkamjan This may be a circular import issue. Read https://stackabuse.com/python-circular-imports/ and try to resolve it.

**Comment by wwongkamjan** [comment]
I will just remove the parameter hint for BaselineBot. Easiest to avoid, However, CI cannot find a module name 'diplomacy_research'. Does CI have diplomacy research folder in there?
https://github.com/ALLAN-DIP/baseline_bots/actions/runs/3146100725/jobs/5114163121

**Comment by trigaten** [comment]
> I will just remove the parameter hint for BaselineBot. Easiest to avoid, However, CI cannot find a module name 'diplomacy_research'. Does CI have diplomacy research folder in there? https://github.com/ALLAN-DIP/baseline_bots/actions/runs/3146100725/jobs/5114163121

It does not. I can try to set that up at a later point. For now, just comment out your test and Ill try to fix this later.

**Comment by wwongkamjan** [comment]
RealPolitik testt removeed

**Comment by trigaten** [comment]
@wwongkamjan Just one last thing: could you add your function call into the `__call__` function of the soa bot in the appropriate place?

**Comment by wwongkamjan** [comment]
done adding best_proposer to SOA

**Comment by trigaten** [comment]
Please run the Makefile

**Review by trigaten — APPROVED** [review]
Thanks!

Resolves #35

**Comment by wwongkamjan** [comment]
this is a weird error. I ran `make` without any problem yesterday

```
login1.ls6(1066)$ make
black src/
make: execvp: black: Permission denied
make: *** [Makefile:2: format_code] Error 127
```

**Comment by trigaten** [comment]
@wwongkamjan I merged regardless. Someone else will run styling soon enough. I do not have a solution for that, unfortunately.

**Pull request merged**